homeassistant/components/conversation/default_agent.py

```diff
@@ -342,10 +342,10 @@ async def async_process(self, user_input: ConversationInput) -> ConversationResu
                 language,
                 assistant=DOMAIN,
             )
-        except intent.NoStatesMatchedError as no_states_error:
+        except intent.MatchFailedError as match_error:
             # Intent was valid, but no entities matched the constraints.
-            error_response_type, error_response_args = _get_no_states_matched_response(
-                no_states_error
+            error_response_type, error_response_args = _get_match_error_response(
```

Comment:
Should `_get_match_error_response` become a method on `intent.MatchFailedError` class ?

Reply:
I don't think it should. The return values are specific to the default agent's responses.